src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -449,10 +449,11 @@ def get_op_handler(self, operation):
         from importlib import import_module
         import types
 
-        from azure.cli.core.profiles import ResourceType
+        from azure.cli.core.profiles import AZURE_API_PROFILES
         from azure.cli.core.profiles._shared import get_versioned_sdk_path
 
-        for rt in ResourceType:
+        resource_types_in_profile = AZURE_API_PROFILES[self.cli_ctx.cloud.profile].keys()
```

Comment:
if `AZURE_API_PROFILE` is a map, you don't need the `.keys()` here. Iterate a map is iterating on its keys.

```
>>> m={'a':1,'b':2}
>>> for i in m:
...   print(i)
... 
a
b
```

Reply:
Any chance a `KeyError` throws here?